rfc/rfc-73/rfc-73.md

```diff
@@ -0,0 +1,416 @@
+taken. However, if a transaction's intent is to ensure that new records of a certain kind don't get added, additional
+mechanisms, like predicate locks, might be needed. If we don't allow snapshot to be refreshed within the same
+transaction, then phantom reads seem improbable (including self-join).
+
+* Textbook phantom read example
+
+  Suppose you have a table named `Employees` with columns `employee_id` and `manager_id`, where the `manager_id` for
+  some employees refers to the `employee_id` of other employees in the same table. One might execute a self-join to
+  retrieve a list of employees and their managers:
+
+  ```plain
+  SELECT e1.employee_id, e2.employee_id AS manager_id
+  FROM Employees e1
+  JOIN Employees e2 ON e1.manager_id = e2.employee_id;
+  ```
+
+  This query essentially matches each employee with their manager, using a self-join on the `Employees` table.
+
+    1. **Transaction A** starts and runs the self-join query to get a list of employees and their respective managers.
+    2. While **Transaction A** is still in progress, **Transaction B** starts and adds a new row to the `Employees`
+       table, inserting a new employee with a manager whose `employee_id` is already in the table.
+    3. **Transaction B** commits its changes.
+    4. If **Transaction A** re-runs the same self-join query, it might see the newly added row, resulting in an
+       additional result in the join output that wasn't there during the initial query. This is a phantom read.
+
+With MVCC and snapshot isolation level, a transaction would continue to see the state of the database as it was when the
+transaction started, even if it re-runs the self-join. This level will prevent the phantom read in this case. However,
+it cannot be guaranteed with read committed.
+
+#### 3\. Read-Write Conflict:
+
+**Definition**: Given a scenario of a concurrent read happening while the transactions are ongoing: A transaction reads
+multiple items, but by the time it finishes reading, another transaction has modified some of the earlier-read items,
+resulting in an inconsistent view.
+
+**Analysis with MVCC**: The data that a transaction reads remains consistent for the duration of the transaction. With
+MVCC and snapshot isolation, the read operation will see a consistent snapshot of the database, depending on when the
+read started. It won't see the uncommitted changes made by the ongoing transactions. So, the read is consistent, and
+there's no anomaly related to the read operation in this setup.
+
+#### 4\. Write-Write Conflict:
+
+**Definition**: Two transactions read the same data, and based on the read, they both decide to modify the data, leading
+to a situation where the final outcome might not be consistent with the intent of either transaction.
+
+**Analysis with MVCC**: This is a **potential problem** even with snapshot isolation. Let's say two transactions, T1 and
```

Comment:
how does this interplay with non blocking concurrency control and conflict resolution proposed in another RFC?